parsers/KR.py

```diff
@@ -1,212 +1,295 @@
+    # print("first_available_date", first_available_date)
+
+    if target_datetime is not None and target_datetime < first_available_date:
+        raise NotImplementedError("This parser is not able to parse dates more than one week in the past.")
+
+    if target_datetime is None:
+        target_datetime = arrow.now(TIMEZONE).datetime
+    
+    target_datetime = time_floor(target_datetime, datetime.timedelta(hours=1))
+
+    # Getting correct row idx
+    target_hour_row_idx = (target_datetime.hour + 23) % 24
+
+    # Getting correct column idx
+    today = arrow.now(TIMEZONE).floor('hour').datetime
+    days_diff = (target_datetime - today).days
+    target_day_col_idx = (7 + days_diff) % 7 if days_diff != 0 else 7
+
+    # print("target_datetime:", target_datetime)
+    # print("target_hour_row_idx:", target_hour_row_idx)
+    # print("days_diff:", days_diff)
+    # print("target_day_col_idx:", target_day_col_idx)
+    # print("today:", today, "target_datetime:", target_datetime, "diff:", target_datetime - today)
+
+    r = session or requests.session()
+    url = PRICE_URL
+
+    response = r.get(url)
+    assert response.status_code == 200
+
+    data = {
+        'zoneKey': zone_key,
+        'datetime': arrow.get(target_datetime, TIMEZONE).datetime,
+        'currency': 'KRW',
+        'price': 0.0,
+        'source': 'new.kpx.or.kr',
+    }
+
+    table_prices = pd.read_html(response.text, header=0)[0]
+    price_cell = table_prices.iloc[target_hour_row_idx, target_day_col_idx]
+
+    if price_cell == 0.0:
+        target_hour_row_idx = 23
+        target_day_col_idx -= 1
+        price_cell = table_prices.iloc[target_hour_row_idx, target_day_col_idx]
+        target_datetime = target_datetime.replace(hour=0, minute=0, second=0, microsecond=0)
+
+    data['price'] = float(price_cell)
+    data['datetime'] = target_datetime
+
+    return data
+
+def get_long_term_prod_data(session=None, target_datetime: datetime.datetime = None) -> dict:
+    target_datetime_formatted_daily = target_datetime.strftime("%Y-%m-%d")
+
+    r = session or requests.session()
+
+    # CSRF token is needed to access the production data
+    r.get(LONG_TERM_PRODUCTION_URL)
+    cookies_dict = r.cookies.get_dict()
+
+    payload = {
+        'mid': 'a10606030000',
+        'device': 'chart',
+        'view_sdate': target_datetime_formatted_daily,
+        'view_edate': target_datetime_formatted_daily,
+        '_csrf': cookies_dict['XSRF-TOKEN'],
+    }
+
+    res = r.post(LONG_TERM_PRODUCTION_URL, payload)
+
+    assert res.status_code == 200
+
+    data = {
+      'zoneKey': 'KR',
+      'datetime': arrow.now(TIMEZONE).datetime,
+      'production': {
+          'biomass': 0.0,
+          'coal':0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': 0.0,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0,
```

Comment:
I'm wondering if it would be better to not set these values to 0 by default, as it means we'll show `solar=0` in the app instead of solar=`?` :)

Reply:
Ok! Now, I just set production and storage to an empty dict. :)